src/domain/tradingview/useTVDatafeed.ts

```diff
@@ -0,0 +1,101 @@
+import { getTokens } from "config/tokens";
+import { timezoneOffset } from "domain/prices";
+import { useChainId } from "lib/chains";
+import { useMemo, useRef } from "react";
+import { getHistoryBars, getLiveBar, supportedResolutions } from "./helper";
+
+const configurationData = {
+  supported_resolutions: Object.keys(supportedResolutions),
+  supports_marks: false,
+  supports_timescale_marks: false,
+  supports_time: true,
+  reset_cache_timeout: 100,
+};
+
+export default function useTVDatafeed() {
+  const { chainId } = useChainId();
+  const intervalRef = useRef<ReturnType<typeof setInterval> | null>();
+  const activeTicker = useRef();
+  return useMemo(() => {
+    return {
+      onReady: (callback) => {
+        setTimeout(() => callback(configurationData));
+      },
+      resolveSymbol: async (symbolName, onSymbolResolvedCallback) => {
+        const symbolInfo = async (symbol) => {
+          const stableTokens = getTokens(chainId)
+            .filter((t) => t.isStable)
+            .map((t) => t.symbol);
+
+          return {
+            name: symbol,
+            type: "crypto",
+            description: symbol + " / USD",
+            ticker: symbol,
+            session: "24x7",
+            minmov: 1,
+            pricescale: 100, // 	or 100
+            timezone: "Etc/UTC",
+            has_intraday: true,
+            has_daily: true,
+            currency_code: "USD",
+            visible_plots_set: true,
+            isStable: stableTokens.includes(symbol),
+          };
+        };
+        const symbol = onSymbolResolvedCallback(await symbolInfo(symbolName));
```

Comment:
looks a bit hack-ish to call some callback here. is it possible to move this callback to e.g. TVChartContainer? or maybe you can explain why it should be here?

Reply:
I have refactored! 

I am using await for fix this error form TV:

 `resolveSymbol` should return result asynchronously. Use `setTimeout` with 0 interval to execute the callback function.